haystack/components/preprocessors/recursive_splitter.py

```diff
@@ -0,0 +1,296 @@
+    >Document(id=..., content: 'AI, in its broadest sense, is intelligence exhibited by machines, particularly computer systems.\n', meta: {'original_id': '65167a9823dd883de577e828ca4fd529e6f7241f0ff616acfce454d808478951', 'split_id': 1, 'split_idx_start': 45, '_split_overlap': []})
+    >Document(id=..., content: 'AI technology is widely used throughout industry, government, and science.', meta: {'original_id': '65167a9823dd883de577e828ca4fd529e6f7241f0ff616acfce454d808478951', 'split_id': 2, 'split_idx_start': 142, '_split_overlap': []})
+    >Document(id=..., content: ' Some high-profile applications include advanced web search engines; recommendation systems; interac...', meta: {'original_id': '65167a9823dd883de577e828ca4fd529e6f7241f0ff616acfce454d808478951', 'split_id': 3, 'split_idx_start': 216, '_split_overlap': []})
+    >]
+    """  # noqa: E501
+
+    def __init__(  # pylint: disable=too-many-positional-arguments
+        self,
+        split_length: int = 200,
+        split_overlap: int = 0,
+        split_units: Literal["word", "char"] = "char",
+        separators: Optional[List[str]] = None,
+        sentence_splitter_params: Optional[Dict[str, Any]] = None,
+    ):
+        """
+        Initializes a RecursiveDocumentSplitter.
+
+        :param split_length: The maximum length of each chunk by default in characters, but can be in words.
+            See the `split_units` parameter.
+        :param split_overlap: The number of characters to overlap between consecutive chunks.
+        :param split_units: The unit of the split_length parameter. It can be either "word" or "char".
+        :param separators: An optional list of separator strings to use for splitting the text. The string
+            separators will be treated as regular expressions unless the separator is "sentence", in that case the
+            text will be split into sentences using a custom sentence tokenizer based on NLTK.
+            See: haystack.components.preprocessors.sentence_tokenizer.SentenceSplitter.
+            If no separators are provided, the default separators ["\n\n", "sentence", "\n", " "] are used.
+
+        :raises ValueError: If the overlap is greater than or equal to the chunk size or if the overlap is negative, or
+                            if any separator is not a string.
+        """
+        self.split_length = split_length
+        self.split_overlap = split_overlap
+        self.split_units = split_units
+        self.separators = separators if separators else ["\n\n", "sentence", "\n", " "]  # default separators
+        self.sentence_tokenizer_params = sentence_splitter_params
+        self._check_params()
+        if "sentence" in self.separators:
+            sentence_splitter_params = sentence_splitter_params or {"keep_white_spaces": True}
+            self.nltk_tokenizer = self._get_custom_sentence_tokenizer(sentence_splitter_params)
```

Comment:
Can we move this to the warm up method like I did in this [PR](https://github.com/deepset-ai/haystack/pull/8646). 

The TLDR is that we want to avoid making external calls to the internet (original motivation of the warm_up method) and also specifically this causes the init method of this component to not be thread safe. Basically when loading NLTK for the first time it tries to download a set of files if they are missing from the local machine.

Reply:
make sense 👍🏽